utils/land-patterns.stanza

```diff
@@ -9,23 +9,120 @@ defpackage ocdb/land-patterns :
     prefix(pad) => def-
   import ocdb/design-vars
 
+;==============================================================================
+;============================== Helpers =======================================
+;==============================================================================
 ; ====== Argument constraint Helpers ===========================================
+; Ensure that a value is even and positive.
 defn ensure-even-positive! (value: Int, name: String):
   if value < 2 or value % 2 == 1:
     fatal("%_ only supports a positive even number of pins." % [name])
 
 ; ====== Reference Label Helpers ===============================================
-public defn ref-label (x:Double, y:Double, a:Anchor) :
-  inside pcb-landpattern :
-    layer(Silkscreen("values", Top)) = Text(">REF", 0.7056, a, loc(x,y))
-    
+; Creates a reference label from pose and anchor
+public defn ref-label (pose:Pose, anchor:Anchor):
+  inside pcb-landpattern:
+    layer(Silkscreen("values", Top)) = Text(">REF", MIN-SILKSCREEN-TEXT-HEIGHT, anchor, pose)
+
+; Creates a reference label at a fixed pose
+public defn ref-label (pose:Pose):
+  ref-label(pose, C)
+
+; DEPRECATED: prefer ref-label(loc(x, y))
+; Creates a reference label at an x,y position and anchor
+public defn ref-label (x:Double, y:Double, anchor:Anchor):
+  ref-label(loc(x, y), anchor)
+
+; DEPRECATED: prefer ref-label(loc(x, y))
+; Creates a reference label at an x,y position with default anchor
 public defn ref-label (x:Double, y:Double) :
   ref-label(x,y,C)
 
+; Create a default reference label
 public defn ref-label () :
-  ref-label(0.0, 0.0)
+  ref-label(loc(0.0, 0.0))
+
+; Create a capsule with normalized rotation
+defn capsule (anchor:Anchor, width:Double, height:Double) -> Capsule:
+  if width >= height:
+    Capsule(anchor, width, height)
+  else:
+    Capsule(width, height, loc(0.0, 0.0, 90.0))
+
+; Create a rounded rectangle with normalized rotation
+defn round-rect (width:Double, height:Double, r:Double) -> RoundedRectangle:
+  if width >= height:
+    RoundedRectangle(width, height, r)
+  else: 
+    RoundedRectangle(height, width, r, loc(0.0, 0.0, 90.0))
+
+; Create a chamfered rectangle with normalized rotation
+defn chamfered (width:Double, height:Double, r:Double) -> ChamferedRectangle:
+  if width >= height:
+    ChamferedRectangle(width, height, r)
+  else:
+    ChamferedRectangle(height, width, r)
+
+;==============================================================================
+;=========================== Geometric Helpers ================================
+;==============================================================================
+; Create a rectangle out of lines
+public defn LineRectangle (w:Double, h:Double, xc:Double, yc:Double, line-w:Double) :
+  val w2 = w * 0.5
+  val h2 = h * 0.5
+  Line(line-w, [
+    Point(xc - w2, yc - h2)
```

Comment:
use `pose = Pose(Point(xc, yc), 0.0, false)` and `apply(pose, Point((- w2), (- h2)))`, ...

Reply:
This is going to be removed in a future iteration, it shows up in the diff because it was relocated in the source file.